camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/CameraPreview.kt

```diff
@@ -15,106 +15,202 @@
  */
 package com.google.jetpackcamera.viewfinder
 
-import android.graphics.Bitmap
+import android.annotation.SuppressLint
 import android.util.Log
+import android.util.Size
 import android.view.Surface
 import android.view.View
 import androidx.camera.core.Preview.SurfaceProvider
 import androidx.camera.core.SurfaceRequest
+import androidx.camera.core.SurfaceRequest.TransformationInfo
 import androidx.camera.view.PreviewView.ImplementationMode
+import androidx.compose.foundation.AndroidEmbeddedExternalSurface
+import androidx.compose.foundation.AndroidExternalSurface
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.produceState
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
-import androidx.compose.runtime.snapshotFlow
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import com.google.jetpackcamera.viewfinder.surface.CombinedSurface
-import com.google.jetpackcamera.viewfinder.surface.CombinedSurfaceEvent
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.setFrom
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import com.google.jetpackcamera.viewfinder.surface.SurfaceTransformationUtil
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.asExecutor
-import kotlinx.coroutines.flow.mapNotNull
 
-private const val TAG = "Preview"
+private const val TAG = "CameraPreview"
 
 @Composable
 fun CameraPreview(
     modifier: Modifier,
     implementationMode: ImplementationMode = ImplementationMode.COMPATIBLE,
     onSurfaceProviderReady: (SurfaceProvider) -> Unit = {},
-    onRequestBitmapReady: (() -> Bitmap?) -> Unit,
+    // TODO(b/267488554): Remove setSurfaceView after implementing an alternative to
+    //  metering point factory
     setSurfaceView: (View) -> Unit
```

Comment:
This method is already unused in this version of the code, why not remove it in this PR?